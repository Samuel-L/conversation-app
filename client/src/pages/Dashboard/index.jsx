import React, { Component } from 'react';

import Grid from '@material-ui/core/Grid';

import CreateConversation from '../../components/CreateConversation';

class Dashboard extends Component {
  state = {};

  render() {
    return (
      <Grid container justify="center">
        <Grid item xs={12} md={6}>
          <CreateConversation />
        </Grid>
      </Grid>
    );
  }
}

export default Dashboard;
