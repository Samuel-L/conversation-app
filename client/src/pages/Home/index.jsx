import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

import { siteTitle } from '../../utils/helpers';
import Login from '../../components/LoginForm';
import Register from '../../components/RegisterForm';

const styles = theme => ({
  root: {
    flex: '1',
    [theme.breakpoints.up('md')]: {
      height: '500px',
    },
  },
});

export class Home extends Component {
  state = {
    showLoginComponent: true,
  }

  handleChangeComponent = () => {
    this.setState({ showLoginComponent: !this.state.showLoginComponent });
  };

  render() {
    const { classes } = this.props;

    return (
      <Grid container alignItems="center" className={classes.root}>
        <Grid item xs={12} md={6}>
          <Typography
            variant="headline"
            align="center"
            gutterBottom
          >
            With { siteTitle } you can send chat with your friends.
          </Typography>
        </Grid>
        <Grid item xs={12} md={6}>
          { this.state.showLoginComponent
            ?
              <Login handleChangeComponent={this.handleChangeComponent} />
            :
              <Register handleChangeComponent={this.handleChangeComponent} />
          }
        </Grid>
      </Grid>
    );
  }
}

Home.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(Home);
