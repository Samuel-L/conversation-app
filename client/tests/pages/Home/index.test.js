import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { Home } from '../../../src/pages/Home';
import store from '../../../src/redux-modules/store.dev';

describe('page: Home', () => {
  describe('rendering', () => {
    it('matches snapshot', () => {
      const classes = { root: '', };
      const tree = renderer.create(
        <Provider store={store}>
          <Home classes={classes} />
        </Provider>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
