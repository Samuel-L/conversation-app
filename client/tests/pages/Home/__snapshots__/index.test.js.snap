// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page: Home rendering matches snapshot 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-39 MuiGrid-grid-md-6-59"
  >
    <h1
      className="MuiTypography-root-92 MuiTypography-headline-97 MuiTypography-gutterBottom-109 MuiTypography-alignCenter-105"
    >
      With 
      Conversationalist
       you can send chat with your friends.
    </h1>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-39 MuiGrid-grid-md-6-59"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-container-1 MuiGrid-justify-xs-flex-end-19"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-39"
        >
          <h1
            className="MuiTypography-root-92 MuiTypography-display1-96"
          >
            Login
          </h1>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-39"
        >
          <div
            className="MuiFormControl-root-119 InputField-formControl-118"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="MuiFormLabel-root-128 MuiInputLabel-root-123 MuiInputLabel-formControl-124 MuiInputLabel-animated-127"
              data-shrink={false}
              htmlFor="username"
            >
              Username
            </label>
            <div
              className="MuiInput-root-133 MuiInput-formControl-134 MuiInput-underline-137"
            >
              <input
                aria-invalid={false}
                aria-required={false}
                className="MuiInput-input-141"
                disabled={false}
                id="username"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="text"
                value=""
              />
            </div>
            <p
              className="MuiFormHelperText-root-146"
              id="username-help-text"
            />
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-12-39"
        >
          <div
            className="MuiFormControl-root-119 InputField-formControl-118"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="MuiFormLabel-root-128 MuiInputLabel-root-123 MuiInputLabel-formControl-124 MuiInputLabel-animated-127"
              data-shrink={false}
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="MuiInput-root-133 MuiInput-formControl-134 MuiInput-underline-137"
            >
              <input
                aria-invalid={false}
                aria-required={false}
                className="MuiInput-input-141 MuiInput-inputType-144"
                disabled={false}
                id="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
            <p
              className="MuiFormHelperText-root-146"
              id="password-help-text"
            />
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-6-33"
        >
          <button
            className="MuiButtonBase-root-172 MuiButton-root-150 MuiButton-textSecondary-153 MuiButton-flat-154 MuiButton-flatSecondary-156 Connect-LoginForm--button-117"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-151"
            >
              Don't have an account? Click here to create one.
            </span>
            <span
              className="MuiTouchRipple-root-175"
            />
          </button>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-6-33 Connect-LoginForm--buttonGrid-116"
        >
          <button
            className="MuiButtonBase-root-172 MuiButton-root-150 MuiButton-contained-159 MuiButton-containedPrimary-160 MuiButton-raised-162 MuiButton-raisedPrimary-163 Connect-LoginForm--button-117"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="submit"
          >
            <span
              className="MuiButton-label-151"
            >
              Login
            </span>
            <span
              className="MuiTouchRipple-root-175"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
